src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -2240,15 +2240,18 @@ FixedwingPositionControl::Run()
 
 				_position_setpoint_previous_valid = PX4_ISFINITE(_pos_sp_triplet.previous.lat)
 								    && PX4_ISFINITE(_pos_sp_triplet.previous.lon)
-								    && PX4_ISFINITE(_pos_sp_triplet.previous.alt);
+								    && PX4_ISFINITE(_pos_sp_triplet.previous.alt)
+								    && _pos_sp_triplet.previous.valid;
```

Comment:
Why is that change needed? Doesn't it break what was patched with https://github.com/PX4/PX4-Autopilot/pull/19942?